chore: documentated non extensiblity of hooks for v5 (#3532)
webpack/webpack.js.org: src/content/api/compilation-hooks.md

The `Compilation` class also extends `Tapable` and provides the following
lifecycle hooks. They can be tapped the same way as compiler hooks:

```js
compilation.hooks.someHook.tap(/* ... */);
```

As with the `compiler`, `tapAsync` and `tapPromise` may also be available
depending on the type of hook.

> W> Note that from __webpack@5__, `hooks` are no longer extendable. Use a `WeakMap` to add custom hooks.

Reviewer: ```suggestion
W> Since webpack 5, `hooks` are no longer extendable. Use a `WeakMap` to add custom hooks.
```
